### PR TITLE
Add FragmentWorker for Off-Policy RL

### PR DESCRIPTION
**Pull request by krzentner**
This is a sampler.Worker which collects fragments of trajectories,
allowing off-policy RL algorithms to update with partial trajectories.

### CODE REVIEW

File: src/garage/sampler/fragment_worker.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        """Take a single time-step in the current rollout.
+
+        Returns:
+            bool: True iff at least one of the paths was completed.
+
+        """
+        prev_obs = np.asarray([frag.last_obs for frag in self._fragments])
+        actions, agent_infos = self.agent.get_actions(prev_obs)
+        completes = [False] * len(self._envs)
+        for i, action in enumerate(actions):
```
Comment (avnishn) [inline src/garage/sampler/fragment_worker.py:104]: why iterate over the number of actions? will this not always be 1 action?

Comment (avnishn) [reply src/garage/sampler/fragment_worker.py:104]: @krzentner bump

Comment (ryanjulian) [reply src/garage/sampler/fragment_worker.py:104]: as the PR description says, it's vectorized.

**Comment by ryanjulian** [comment]
@Mergifyio rebase

**Comment by krzentner** [comment]
There are still some test failures with this change (specifically, the off-policy TF algos report low performance in tests, despite working normally when I run examples).

**Pull request merged**